### PR TITLE
Header section

### PR DESCRIPTION
**Pull request by jorgearellano19**
This PR contains the header section. 

Issue associated: #99

### CODE REVIEW

**Comment by eduardorr97** [comment]
Code looks good, but I have some comments:
- The navbar color in the designs is always white so lets use that
- Make sure the Search bar sticks to the far right like in the designs (try using just flex, avoiding absolute positioning and so)
- Needs to adjust the breakpoint for the hamburguer (around 1110px should be enough)
- Spacing between `nav-item`s should be somewhere around 1rem
- `nav-item`'s bottom border when they are active should be 2px
- Notice that the `nav-item` arrow flips when they are active
- The first time you click a `nav-item` you get a warning in the console:
```
 validateDOMNesting(...):  <a> cannot appear as a descendant of <a>.
```
- Let's add at least a # as `href` to the links that we don't have the actual `href`s

File: web/src/components/Layout/components/Header/components/Option/Option.tsx [hunk 1]
```diff
@@ -1,29 +1,50 @@
...
-          </NavDropdown.Item>
-        ))}
+      <NavDropdown onToggle={onToggle} className={dropdownClass} title={title}>
+        {collection && (
+          <OptionDropdownCollection
+            sanityCollection={sanityCollection}
+            collection={collection}
+          />
+        )}
+        {items && <OptionDropdownItems items={items} />}
```
Comment (eduardorr97) [inline web/src/components/Layout/components/Header/components/Option/Option.tsx:47]: So this works just fine, however, typescript is complaining about the `onToggle` prop, we have to find a way to fix that, may be in another issue?

Also this whole section is looking a bit obscure, can you refactor it to something like this or maybe a `switch case`:

```
if (type === header_option_types.link) {
    return <Nav.Link className={s.buttonLink}>{title}</Nav.Link>
  } else if (type === header_option_types.dropdown) {
    return (
      <NavDropdown onToggle={onToggle} className={dropdownClass} title={title}>
        {collection && (
          <OptionDropdownCollection
            sanityCollection={sanityCollection}
            collection={collection}
          />
        )}
        {items && <OptionDropdownItems items={items} />}
      </NavDropdown>
    )
  } else return <></>
```

File: web/src/components/Layout/components/Header/components/Option/Option.tsx [hunk 2]
```diff
@@ -1,29 +1,50 @@
...
+  title,
+  type,
+  sanityCollection,
+  collection,
+  items,
+}) => {
+  const [dropdownClass, setDropdownClass] = useState(classnames(s.dropdown))
+  const onToggle = (isOpen, event, metadata) => {
+    setDropdownClass(
+      isOpen ? classnames(s.activeDropdown, s.dropdown) : classnames(s.dropdown)
```
Comment (eduardorr97) [inline web/src/components/Layout/components/Header/components/Option/Option.tsx:33]: add a `type` to `isOpen`, if `event` and `metadata` wont be used, remove them

**Comment by eduardorr97** [comment]
This is how it looks when you are on a small screen and open the menu and one of the dropdowns

Screenshot from 2020-08-27 16-57-01

This can be fixed probably with just some `z-index` in `css`

File: web/src/components/Layout/components/Header/components/OptionDropdownCollection/OptionDropdownCollection.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      <Col xs={12} md={4}>
+        <div className={s.title}>Products</div>
+        <Row>
+          {products.map((product, index) => (
+            <Col key={index} xs={12}>
+              <NavDropdown.Item bsPrefix={s.dropdownItem} role="navigation">
+                <Link to={`/product/${product.handle}`}>
+                  <p className={s.productTitle}>{product.title}</p>
+                </Link>
+              </NavDropdown.Item>
```
Comment (eduardorr97) [inline web/src/components/Layout/components/Header/components/OptionDropdownCollection/OptionDropdownCollection.tsx:28]: Do we really need all 3 this elements? I think we can get away with just the link

```
<Link to={`/product/${product.handle}`}>{product.title}</Link>
```

File: web/src/components/Layout/components/Header/components/OptionDropdownCollection/OptionDropdownCollection.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      <Col xs={12} md={4}>
+        <div className={s.title}>Resources</div>
+        <Row>
+          {faqs.map((faq, index) => (
+            <Col key={index} xs={12}>
+              <NavDropdown.Item bsPrefix={s.dropdownItem} role="navigation">
+                <Link to={`/collection/${collection.handle}`}>
+                  <p className={s.productTitle}>{faq.question}</p>
+                </Link>
+              </NavDropdown.Item>
```
Comment (eduardorr97) [inline web/src/components/Layout/components/Header/components/OptionDropdownCollection/OptionDropdownCollection.tsx:42]: Same as above to me

File: web/src/components/Layout/components/Header/components/OptionDropdownCollection/OptionDropdownCollection.tsx (hunk 3 above)
Comment (jorgearellano19) [reply web/src/components/Layout/components/Header/components/OptionDropdownCollection/OptionDropdownCollection.tsx]: The Navdropdown.Item is there for getting the behaivour of a dropdown (if we click on an Item, the dropdown closes). I solved with the prop as="div" and letting just the <Link> component.

**Pull request merged**